### DISCUSSION TITLE
Can I connect to an existing server instance?

### DISCUSSION THREAD

**Discussion by prathik-quartic** [FreeOpcUa/opcua-asyncio#575] [Q&A] [answered]
If I had a server that was already created on a machine, with objects and variables defined, can I search and access this server instance on that machine? 
I want to access this particular server instance and update the values of the variables present in it.

**Comment by swamper123** [comment, marked as answer]
There is a tool for discovery services in tools.py to find servers in your network (tbh: I never used that so far), but if this is your only server on the machine, you can get access as normal (via endpoint). [See our example](https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples/client-minimal.py).

If you got a connection, you can browse the node and execute write_value() (keep in mind, to use variants, instead of normal numbers/strings.).

**Reply by prathik-quartic** [reply]
Thank you for the response. I did try that. 
```
async def main():
    url = "opc.tcp://0.0.0.0:4840/freeopcua/server/"
    async with Client(url=url) as client:
        print("Root node is: %r", client.nodes.root)
        print("Objects node is: %r", client.nodes.objects)
        print("Children of root are: %r", await client.nodes.root.get_children())
if __name__ == '__main__':
    asyncio.run(main(), debug=True)
```

I got this execption:

```
Traceback (most recent call last):
  File "client_spectral_data_generator.py", line 35, in <module>
    asyncio.run(main(), debug=True)
  File "/usr/lib/python3.8/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "client_spectral_data_generator.py", line 19, in main
    async with Client(url=url) as client:
AttributeError: __aexit

```

**Reply by swamper123** [reply]
I can remember that the IP: 0.0.0.0 often causes some issues...

Could you try following constelation, perhaps we get closer to the root of problems:

```python

async def main():
    url = "opc.tcp://0.0.0.0:4840/freeopcua/server/"
    client = Client(url=url)
    client.connect()
    print("Root node is: %r", client.nodes.root)
    print("Objects node is: %r", client.nodes.objects)
    print("Children of root are: %r", await client.nodes.root.get_children())
    client.disconnect()
    print("End of Client connection")
if __name__ == '__main__':
    asyncio.run(main(), debug=True)

**Reply by prathik-quartic** [reply]
Yep. Worked. Thank you so much:)

**Reply by swamper123** [reply]
Hmm... strange to be honest.... ^^
I just saw that connect and disconnect are coroutines, so an await should be added before clien.connect()/.disconnect().
And entering the statement "async with Client ..." doesn't do anything else as the connect and disconnect method.
